Dockerfile

```diff
@@ -68,27 +66,39 @@ RUN apt-get update \
   && rm -rf /var/lib/apt/lists/*
 
 ARG SPARK_VERSION
-ARG SPARK_VARIANT
+ARG HADOOP_VERSION
 ARG SPARK_PY4J
 
+ENV HADOOP_HOME /opt/hadoop
+ENV PATH ${PATH}:${HADOOP_HOME}/bin:${HADOOP_HOME}/sbin
+ENV HADOOP_CONF_DIR ${HADOOP_HOME}/etc/hadoop
+ENV HDFS_CONF_DIR ${HADOOP_CONF_DIR}
+ENV YARN_CONF_DIR ${HADOOP_CONF_DIR}
+ENV HIVE_CONF_DIR ${HADOOP_CONF_DIR}
+
 ENV SPARK_HOME=/opt/spark-${SPARK_VERSION}
 ENV PATH=$PATH:${SPARK_HOME}/bin
 ENV PYTHONPATH=${SPARK_HOME}/${SPARK_PY4J}:${SPARK_HOME}/python
 ENV PYSPARK_SUBMIT_ARGS="--driver-memory 8g --py-files ${SPARK_HOME}/python/lib/pyspark.zip pyspark-shell"
 
 # Download Spark
 ARG SPARK_EXTRACT_LOC=/sparkbin
-RUN set -eux && \
+RUN set -eux && \\
+    curl http://www-us.apache.org/dist/hadoop/common/hadoop-${HADOOP_VERSION}/hadoop-${HADOOP_VERSION}.tar.gz | \
+    tar -xz -C /opt/ && \
+    mv /opt/hadoop-${HADOOP_VERSION} /opt/hadoop && \
     mkdir -p ${SPARK_EXTRACT_LOC} && \
-    curl https://www.mirrorservice.org/sites/ftp.apache.org/spark/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-${SPARK_VARIANT}.tgz |\
+    curl https://www.mirrorservice.org/sites/ftp.apache.org/spark/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-hadoop${HADOOP_VERSION:0:3}.tgz | \
```

Comment:
Can use the Hadoop-provided variant to keep bundle small. But need to update spark-submit.sh subsequently

Reply:
*-env.sh

Reply:
I'm not sure what you are referring to about hadoop provided variant.

Reply:
`export SPARK_DIST_CLASSPATH=$(hadoop classpath)` (the content in spark-env.sh) is already in `entrypoint.sh`

Reply:
Fetching spark binaries without hadoop now.

Reply:
Since the spark binary without hadoop pre-built does not come with hive, im just going to fetch this now.